integreat_cms/cms/models/media/media_file.py

```diff
@@ -259,6 +259,39 @@ def content_usages(self) -> QuerySet:
         """
         return Link.objects.filter(url__url=self.url).distinct("object_id")
 
+    @cached_property
+    def past_event_usages(self) -> QuerySet:
+        """
+        Count in how many past events this file is used
+
+        :return: count of usages in past events
+        """
+        return self.events.filter(end__lt=timezone.now().date())
+
+    @cached_property
+    def is_only_used_in_past_events(self) -> QuerySet:
+        """
+        Check if a media file is used in past events only
+
+        :return: if a media file is only used in past events
+        """
+        usage_counts = [
+            self.past_event_usages.count(),
+            (len(self.icon_usages) + self.content_usages.count()),
+            self.events.count(),
+        ]
+
+        return all(c == usage_counts[0] for c in usage_counts)
```

Comment:
Sorry, my suggestion introduced another bug/edge case 🙈 
`past_event_usages` finds only events in which the media file is used as icon and does not find use case as content 🙈 

So, I'd suggest to adjust `is_only_used_in_past_events` as below

```suggestion
        past_event_translation_ids = [
            event_translation.id
            for event_translation in EventTranslation.objects.filter(
                event__end__lt=timezone.now().date()
            )
        ]

        return (
            len(self.icon_usages) == self.past_event_usages.count()
            and not self.content_usages.exclude(
                content_type__id=ContentType.objects.get(model="eventtranslation").id,
                object_id__in=past_event_translation_ids,
            ).count()
            > 0
        )
```

if this is applied, `from ..events.event_translation import EventTranslation` and `from django.contrib.contenttypes.models import ContentType` need to be imported